src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -1610,10 +1614,61 @@ public static int Sum(this IEnumerable<int> source)
             int sum = 0;
             checked
             {
-                foreach (int v in source) sum += v;
+                if (Vector.IsHardwareAccelerated && source.Count() >= SIMD_THRESHOLD)
```

Comment:
The _Count_ method would probably introduce the additional iteration over the _source_, which is not intended here.